curves/src/pasta/curves/pallas.rs

```diff
@@ -1,70 +1,67 @@
-use crate::pasta::*;
 use ark_ec::{
-    models::short_weierstrass_jacobian::{GroupAffine, GroupProjective},
-    ModelParameters, SWModelParameters,
+    models::CurveConfig,
+    short_weierstrass::{self as sw, SWCurveConfig},
 };
-use ark_ff::{field_new, Zero};
+use ark_ff::{Field, MontFp, Zero};
 
-#[derive(Debug, Clone, Default, PartialEq, Eq)]
-pub struct PallasParameters;
+use crate::pasta::{fp::Fp, fq::Fq};
 
-impl ModelParameters for PallasParameters {
+#[derive(Debug, Copy, Clone, Default, PartialEq, Eq)]
+pub struct PallasConfig;
+
+impl CurveConfig for PallasConfig {
     type BaseField = Fp;
     type ScalarField = Fq;
+
+    /// COFACTOR = 1
+    const COFACTOR: &'static [u64] = &[0x1];
+
+    /// COFACTOR_INV = 1
+    const COFACTOR_INV: Fq = Fq::ONE;
 }
 
-pub type Pallas = GroupAffine<PallasParameters>;
-pub type ProjectivePallas = GroupProjective<PallasParameters>;
+pub type Affine = sw::Affine<PallasConfig>;
```

Comment:
I think Pallas and Vesta implementations should actually be as close to identical as possible. Perhaps, even parameterized over base and scalar fields and the generators. IMO, this would make for a nice follow-up issue.

Reply:
Check out what it looks like now 😅